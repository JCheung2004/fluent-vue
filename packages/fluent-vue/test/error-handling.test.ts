import { createLocalVue, mount } from '@vue/test-utils'

import { FluentBundle, FluentResource } from '@fluent/bundle'
import ftl from '@fluent/dedent'

import FluentVue from '../src'

describe('vue integration', () => {
  const localVue = createLocalVue()
  localVue.use(FluentVue)

  const bundle = new FluentBundle('en-US')

  bundle.addResource(
    new FluentResource(ftl`
    message = Hello, { $name }!
    sub-message = Hi, { $name }
  `)
  )

  const fluent = new FluentVue({
    bundles: [bundle]
  })

  const options = {
    fluent,
    localVue
  }

  it('warns about missing translation', () => {
    // Arrange
    const component = {
      template: "<div>{{ $t('message-not-found') }}</div>"
    }

    const warn = jest.fn()
    console.warn = warn

    // Act
    const mounted = mount(component, options)

    // Assert
    expect(mounted).toMatchSnapshot()
    expect(warn).toHaveBeenCalledTimes(1)
  })

  it('warns about fluent errors', () => {
    // Arrange
    bundle.addResource(
      new FluentResource(ftl`
      message = { NUMBER($arg) }
      `)
    )

    const component = {
      template: "<div>{{ $t('message') }}</div>"
    }

    const warn = jest.fn()
    console.warn = warn

    // Act
    const mounted = mount(component, options)

    // Assert
    expect(mounted).toMatchSnapshot()
    expect(warn).toHaveBeenCalledTimes(1)
  })
})
