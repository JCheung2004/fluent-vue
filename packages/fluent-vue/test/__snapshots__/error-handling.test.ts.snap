// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue integration warns about fluent errors 1`] = `<div>{NUMBER($arg)}</div>`;

exports[`vue integration warns about missing translation 1`] = `<div>message-not-found</div>`;
